#labels Featured
# Installation instructions

=== MATLAB version ===
The code consists of a set of MATLAB scripts. No installation is required.
The MATLAB code was developed using MATLAB R2011b. 
The R plotting code was developed using R versions 2.13-2.15 and requires the
ggplot2 and reshape libraries.
All the code was developed on 64-bit Linux.

The code should be compatible with older (but reasonably new) vesions 
of MATLAB and R and with other platforms, but we have only tested it in the settings mentioned above. Please email us if you run the code in other platforms, so we can update the requirements.

=== Other distributions ===
In the near future, we plan to create MATLAB compiled code which you can run with the MATLAB compiler (i.e. without a full MATLAB installation). We're also developing a Python version of CAGT, but this won't be available soon.

=== Recommended software ===
The following code is not required, but you might find it useful in creating the input files for CAGT:
  * [http://code.google.com/p/align2rawsignal/ align2rawsignal]: Reads files with mapped reads (eg. BAM) and creates tracks of signal along the genome. It can apply smoothing, normalization, or other filters.
  * [http://code.google.com/p/extractsignal/ extractSignal]: Starting from a genomewide signal and a set of loci of interest, it extracts the signal around the loci.